src/js/components/DataTable/ExpanderCell.js

```diff
@@ -35,11 +47,45 @@ const ExpanderControl = ({ context, expanded, onToggle, pad, ...rest }) => {
   );
 
   if (onToggle) {
+    const expandText =
+      messages?.expand ||
+      format({
+        id: 'dataTable.expand',
+        messages: {
+          'dataTable.expand': 'expand',
+        },
+      });
+
+    const collapseText =
+      messages?.collapse ||
+      format({
+        id: 'dataTable.collapse',
+        messages: {
+          'dataTable.collapse': 'collapse',
+        },
+      });
+
+    let a11yTitle;
+
+    if (context === 'header') {
+      a11yTitle = expanded ? 'collapse all' : 'expand all';
+    } else {
+      a11yTitle = expanded ? collapseText : expandText;
+
+      if (expandAriaLabel) {
+        const ariaLabel =
+          typeof expandAriaLabel === 'function'
+            ? expandAriaLabel(row, groupKey)
+            : expandAriaLabel;
```

Comment:
If expandAriaLabel isn't a function it seems like `ariaLabel` should just be undefined